enhancements/installer/bootstrap-ignition-shim-asset.md

```diff
@@ -0,0 +1,123 @@
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift/docs]
+
+## Summary
+
+There are a few platforms that use an Ignition shim for their bootstrap machine. The shim
+is essentially a pointer to the main config that is stored on a remote server, which helps
+us avoid problems with instance metadata size limits on our platform. Previously, we were doing
+this in Terraform using the terraform Ignition Provider, however the repo has fallen extremely out
+of date, making us unable to add additional CA Bundles to our nodes. This means that clients that
+use self signed certificates are unable to use our IPI installer, and can not upgrade to openshift 4 on Openstack.
+We considered both updating the
+terraform provider and creating a new installer asset as possible solutions, and ultimately decided
+that the best solution for our team is to create the installer asset. We plan to create an ignition
+asset similar to the [node asset](https://github.com/openshift/installer/blob/master/pkg/asset/ignition/machine/node.go)
+for bootstrap ignition shims that other platforms can **opt in** to.
+
+## Motivation
+
+We spent a lot of time looking at the Terraform Ignition Provider to determine if we could
+get involved in the community there to solve our problem. However we had a number of hangups
+with this. First and foremost, that repo has not seen code merged since May 2019. We also noticed
+a number of valid and mergable pull requests waiting for approval for months.
+It seems as if there aren't any maintainers, and we were also unable to reach the contributers.
+The CA Bundle bug is preventing a considerable portion of our IPI users
+from being able to upgrade to OpenShift 4 on OpenStack, and we need this fix in 4.3, so this was rather
+dissuading. Lastly, we discovered that for our UPI solution, we wont have a way to generate the bootstrap
+ignition shim if we continue generating it in terraform. We would like to generate the ignition configs for
+UPI with the command `openshift-install create ignition-configs` so that Ignition configs for both UPI and IPI
+are created the same way.
+
+### Goals
+
+Our goal is to create an installer asset, written in Go, that creates an Ignition
+shim for the bootstrap node. This shim should contain only the data
+required to succesfully fetch the bootstrap Ignition config from its endpoint.
+All further customizations should be added to the bootstrap Ignition config, not
+the bootstrap shim. The design of this asset should be similar to the node ignition
+asset so that other platforms can opt into using it, and add their own customizations.
+
+## Proposal
+
+We are proposing to create a new installer asset in `installer/pkg/asset/ignition/bootstrap/shim.go`.
```

Comment:
Have you guys considered using string manipulation to create a 2.3.0 schema ign json file that has the CAbundle attached??

That would solve the IPI Openstack problem. and as for other platforms there isn't a need for this on cloud and i think for UPI we are more suited to helping our users creating a ign using bootstrap.ign using existing tools or making sure our UPI templates can safely generate those.

Reply:
I was not aware of this solution, we havn't considered it.

Reply:
Would this also allow for Spec3 support? Possibly related: @vrutkovs had to remove the terraform plugin for now (and with it all of Azure support) in the Spec3 PR on Installer: https://github.com/openshift/installer/pull/2548

Reply:
That doesn't affect spec3 support much. Terraform provider still needs to be forked so that it could set ign version 3.0.0 in pointer ignition set in machine's user-data

Reply:
Maybe in 4.4, the OpenStack and Azure teams could team up on making an OpenShift fork of the terraform-ignition-provider

Reply:
> Maybe in 4.4, the OpenStack and Azure teams could team up on making an OpenShift fork of the terraform-ignition-provider

that would be brilliant because folks who started with 4.0 might have already got some investment in other wrappers around and moving away from it would cause disruption for sure